### PR TITLE
MOFED-LTS; MLNX_OFED_LINUX-4.9-3.1.5.0; HPC-x v2.7.0

### PR DESCRIPTION
**Pull request by abhamidipati-msft**

### CODE REVIEW

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - OpenUCX
+- Optimized librares
+  - AMD Blis
+  - AMD FFTW
+  - AMD Flame
+  - Intel MKL
+- GPU Drivers
+  - Nvidia GPU Driver
+- SHARP Daemon (sharpd)
+- GDR Copy
```
Comment (jithinjosepkl) [inline ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/README.md:23]: Please remove SHARP, GDR copy.

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install_dcgm.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```
Comment (jithinjosepkl) [inline ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install_dcgm.sh:1]: Any difference for dcgm as compared to regular Ubuntu-HPC image? If not, should we use a common script?

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install_nvidiagpudriver.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+set -ex
+
+$COMMON_DIR/install_nvidiagpudriver.sh
+
+# Install gdrcopy
```
Comment (jithinjosepkl) [inline ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install_nvidiagpudriver.sh:6]: GDR copy is not needed.

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install_mpis.sh [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+TARBALL="hpcx-${HPCX_VERSION}-gcc-MLNX_OFED_LINUX-${MLNX_OFED_VERSION}-ubuntu18.04-x86_64.tbz"
+HPCX_DOWNLOAD_URL=https://azhpcstor.blob.core.windows.net/azhpc-images-store/${TARBALL}
+HPCX_FOLDER=$(basename ${HPCX_DOWNLOAD_URL} .tbz)
+
+$COMMON_DIR/download_and_verify.sh $HPCX_DOWNLOAD_URL "83f03e2d01cb1e4198e50ed9bdadb742aebf5d247369071bc911096824147d7a"
+tar -xvf ${TARBALL}
+mv ${HPCX_FOLDER} ${INSTALL_PREFIX}
+HPCX_PATH=${INSTALL_PREFIX}/${HPCX_FOLDER}
+
+# Enable Sharpd
```
Comment (jithinjosepkl) [inline ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install_mpis.sh:24]: we can remove this.

**Review by jithinjosepkl — CHANGES_REQUESTED** [review]

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install_nvidiagpudriver.sh [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+sudo apt-mark hold gdrdrv-dkms
+sudo dpkg -i libgdrapi_${GDRCOPY_VERSION}-1_amd64.Ubuntu18_04.deb
+sudo apt-mark hold libgdrapi
+sudo dpkg -i gdrcopy-tests_${GDRCOPY_VERSION}-1_amd64.Ubuntu18_04.deb
+sudo apt-mark hold gdrcopy-tests
+sudo dpkg -i gdrcopy_${GDRCOPY_VERSION}-1_amd64.Ubuntu18_04.deb
+sudo apt-mark hold gdrcopy
+popd
+
+# Install nvidia fabric manager (required for ND96asr_v4)
```
[the remark this answers is not in the document]
Comment (abhamidipati0614) [reply ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install_nvidiagpudriver.sh]: NVIDIA fabric manager is not needed. Updated the PR.

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/README.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Ubuntu 18.04 HPC Image
+
```
[the remark this answers is not in the document]
Comment (abhamidipati0614) [reply ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/README.md:3]: Added

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install_dcgm.sh (hunk 2 above)
Comment (abhamidipati0614) [reply ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install_dcgm.sh:1]: dcgm is identified to have similar structure for Ubuntu. Created a common script.

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install_mpis.sh (hunk 4 above)
Comment (abhamidipati0614) [reply ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install_mpis.sh]: Removed

**Pull request merged**